main.go

```diff
@@ -142,6 +143,55 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 	if resp.StatusCode != http.StatusOK {
 		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
 	}
+	mediatype, params, err := mime.ParseMediaType(resp.Header.Get("Content-Type"))
+	if err == nil && mediatype == "application/vnd.google.protobuf" &&
+		params["encoding"] == "delimited" &&
+		params["proto"] == "io.prometheus.client.MetricFamily" {
+		for {
+			mf := &dto.MetricFamily{}
+			if _, err = pbutil.ReadDelimited(resp.Body, mf); err != nil {
+				if err == io.EOF {
+					break
+				}
+				log.Fatalln("reading metric family protocol buffer failed:", err)
+			}
+			ch <- mf
+		}
+	} else {
+		// We could do further content-type checks here, but the
+		// fallback for now will anyway be the text format
+		// version 0.0.4, so just go for it and see if it works.
+		var parser expfmt.TextParser
+		metricFamilies, err := parser.TextToMetricFamilies(resp.Body)
+		if err != nil {
+			log.Fatalln("reading text format failed:", err)
+		}
+		for _, mf := range metricFamilies {
+			ch <- mf
+		}
+	}
+}
+
+func fetchMetricFamiliesWithAuth(url string, ch chan<- *dto.MetricFamily, certFile string, keyFile string) {
```

Comment:
How does this method leverage the old method?  Sorry for being a `go` nube, but I would think we should just enhance the original method to take the additional TLS parameters.